server/src/main/java/org/elasticsearch/common/network/NetworkModule.java

```diff
@@ -94,6 +98,18 @@
             new NamedWriteableRegistry.Entry(Task.Status.class, ReplicationTask.Status.NAME, ReplicationTask.Status::new));
         namedWriteables.add(
             new NamedWriteableRegistry.Entry(Task.Status.class, RawTaskStatus.NAME, RawTaskStatus::new));
+        namedWriteables.add(
```

Comment:
I think that `ClusterModule` is a better place to register these classes, because this class already has the majority (I think all of them) default custom metadata registrations and in my mind `NetworkModule` is more for real network things like http and transport.

Reply:
I agree.